### PR TITLE
feat: add to_dict() and get_headers() methods to Headers

### PR DESCRIPTION
**Pull request by sansyrox**
## Summary

- Adds `Headers.to_dict()` which returns a flat `dict[str, str]` with comma-joined values for duplicate headers, matching standard HTTP semantics.
- Adds Python type stubs for both `to_dict()` and `get_headers()` in `robyn.pyi`.
- Adds unit tests covering single values, duplicates, empty headers, and `get_headers()` list preservation.

Supersedes #1184

## Test plan

- [x] `test_headers_to_dict` — basic single-value headers
- [x] `test_headers_to_dict_with_duplicates` — comma-joining of multi-value headers
- [x] `test_headers_get_headers` — list preservation via `get_headers()`
- [x] `test_headers_to_dict_empty` — empty headers edge case

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * Added `get_headers()` method to retrieve all headers grouped by name with values as lists.
  * Added `to_dict()` method to retrieve headers as a flattened dictionary, with multiple values concatenated by commas.

* **Tests**
  * Added unit tests for both new header retrieval methods, including edge cases for duplicate and missing headers.

### CODE REVIEW

File: src/types/headers.rs [hunk 1]
```diff
@@ -94,6 +94,20 @@ impl Headers {
...
 
+    pub fn to_dict(&self, py: Python) -> Py<PyDict> {
+        let dict = PyDict::new(py);
+        for iter in self.headers.iter() {
+            let (key, values) = iter.pair();
+            let joined_value = if values.len() == 1 {
+                values[0].clone()
+            } else {
+                values.join(",")
+            };
```
[the remark this answers is not in the document]
Comment (sansyrox) [reply src/types/headers.rs:105]: Resolve — by design. to_dict() is a convenience flattened view. get_headers() preserves full multi-value fidelity for headers like set-cookie.

**Comment by sansyrox** [comment]
Closing this in favour of a clean, consolidated implementation. The competing to_dict PRs (#1376, #1184) will be superseded by a single fresh PR.
